run.sh

```diff
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-cd `dirname $0`
+cd "$(dirname "$0")"
```

Comment:
using `$(...)` is not strict Bourne compatible, although dash supports it since dash is POSIX 1003.2 compatible. Which is probably good enough.